### PR TITLE
PIPE2D-296: Add SpectrumSet.plot method

### PR DESCRIPTION
**Pull request by PaulPrice**
This involves a bit of refactoring of Spectrum.plot so they can share
code. Renamed the plotBackground and plotReferenceLines kwargs to
doBackground and doReferenceLines.

SpectrumSet.plot doesn't have all the features that Spectrum.plot does,
as I decided that plotting the background and reference lines would just
make the plots too confusing.

### CODE REVIEW

File: tests/test_Spectrum.py [hunk 1]
```diff
@@ -382,6 +398,35 @@ def testReadWriteFits(self):
...
+            spectra.plot(numRows=1, filename=filename)
+        # Test one row, check return values
+        with lsst.utils.tests.getTempFilePath(".png") as filename:
+            figure, axes = spectra.plot(numRows=1)
+            with self.assertRaises(TypeError):
+                axes[0]
+            figure.savefig(filename)
+
 
 class TestMemory(lsst.utils.tests.MemoryTestCase):
```
Comment (hassanxp) [inline tests/test_Spectrum.py]: (Outside the context of this PR I know) but why is this class empty?

File: tests/test_Spectrum.py [hunk 2]
```diff
@@ -236,8 +236,24 @@ def testPlot(self):
         import matplotlib.pyplot as plt
         plt.switch_backend("agg")  # In case someone has loaded a different backend that will cause trouble
         spectrum = self.makeSpectrum()
+        # Write directly to file
         with lsst.utils.tests.getTempFilePath(".png") as filename:
-            spectrum.plot(numRows=4, plotBackground=True, plotReferenceLines=True, filename=filename)
+            spectrum.plot(numRows=4, doBackground=True, doReferenceLines=True, filename=filename)
+        # Check return values
+        with lsst.utils.tests.getTempFilePath(".png") as filename:
```
Comment (hassanxp) [inline tests/test_Spectrum.py]: .png hardcoded? Can there be support for other image types (JPG, PDF)?

File: tests/test_Spectrum.py (hunk 1 above)
Comment (PaulPrice) [reply tests/test_Spectrum.py]: It's so that pytest can find it. It's part of the LSST test suite boilerplate.

File: tests/test_Spectrum.py [hunk 3]
```diff
@@ -236,8 +236,24 @@ def testPlot(self):
         import matplotlib.pyplot as plt
         plt.switch_backend("agg")  # In case someone has loaded a different backend that will cause trouble
         spectrum = self.makeSpectrum()
+        # Write directly to file
         with lsst.utils.tests.getTempFilePath(".png") as filename:
```
Comment (hassanxp) [inline tests/test_Spectrum.py]: ".png" repeated across code. Define a constant?

File: python/pfs/drp/stella/spectraContinued.py [hunk 4]
```diff
@@ -20,53 +20,136 @@
...
+                       badReference=badReference)
+
+        if filename is not None:
+            figure.savefig(filename, bbox_inches='tight')
+            plt.close(figure)
+        return figure, axes
+
+    def _plotImpl(self, axes, division, doBackground=False, doReferenceLines=False,
+                  badReference=BAD_REFERENCE, fluxStyle=None, backgroundStyle=None):
+        """Private implementation for plotting spectrum
```
Comment (hassanxp) [inline python/pfs/drp/stella/spectraContinued.py]: Does this really need to be private?

File: tests/test_Spectrum.py (hunk 1 above)
Comment (hassanxp) [reply tests/test_Spectrum.py]: Ok thanks

**Pull request merged**